example3/API/MetricSource.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+namespace OpenTelmetry.Api
+{
+    public abstract class MetricSource
+    {
+        protected MetricSource oldSource = null;
+
+        public MetricSource()
+        {
+        }
+
+        public void RegisterSDK()
+        {
+            MetricBase.RegisterSDK(this);
+        }
+
+        /// <summary>
+        /// Let SDK know when new counters are being created
+        /// </summary>
+        public abstract bool OnCreate(MetricBase counter, LabelSet labels);

```

Comment:
Would this assume the SDK is injected before the counters are being created?

Reply:
I would prefer to allow the SDK to attach whenever.  But you make a good point, about what we do with these OnCreate events if an SDK is currently not attached.  Or if we do anything after a SDK is attached for previously created item.

Reply:
This issue is no longer applicable with current implementation of MetricSource